sdks/unity/AgonesSdkClient.cs

```diff
@@ -0,0 +1,179 @@
+        /// record will be moved to the Ready state, and the details for its public address
+        /// and connection port will be populated.
+        /// </summary>
+        public void Ready()
+        {
+            AgonesSdkClient.EmptyPost("http://localhost:59358/ready");
+        }
+
+        /// <summary>
+        /// This sends a single ping to designate that the Game Server is alive and healthy.
+        /// Failure to send pings within the configured thresholds will result in the GameServer
+        /// being marked as Unhealthy.
+        /// </summary>
+        public void Health()
+        {
+            AgonesSdkClient.EmptyPost("http://localhost:59358/health");
+        }
+
+        /// <summary>
+        /// This tells Agones to shut down the currently running game server. The GameServer
+        /// state will be set Shutdown and the backing Pod will be deleted, if they have not
+        /// shut themselves down already.
+        /// </summary>
+        public void Shutdown()
+        {
+            AgonesSdkClient.EmptyPost("http://localhost:59358/shutdown");
+        }
+
+        /// <summary>
+        /// Request agones to set a label key/value pair
+        /// </summary>
+        /// <param name="key">The key name</param>
+        /// <param name="value">The value</param>
+        public void SetLabel(string key, string value)
+        {
+            this.SetProperty(key, value, "http://localhost:59358/metadata/label");
+        }
+
+        /// <summary>
+        /// Requests Agones to set an annotation key/value pair
+        /// </summary>
+        /// <param name="key">The key value</param>
+        /// <param name="value">The value</param>
+        public void SetAnnotation(string key, string value)
+        {
+            this.SetProperty(key, value, "http://localhost:59358/metadata/annotation");
+        }
+
+        /// <summary>
+        /// Co-routine that calls the health endpoint at UpdateHertz rate
+        /// </summary>
+        private IEnumerator UpdateLoop()
+        {
+            while (true)
+            {
+                this.Health();
+                yield return this.wait;
+            }
+        }
+
+        /// <summary>
+        /// Makes a call to set a key/value pair
+        /// </summary>
+        /// <param name="key">The key name</param>
+        /// <param name="value">The value</param>
+        /// <param name="uri">The URI to post the key/value pair</param>
+        private void SetProperty(string key, string value, string uri)
+        {
+            KeyValueMessage msg = new KeyValueMessage()
+            {
+                key = key,
+                value = value
+            };
+
+            string payload = JsonConvert.SerializeObject(msg);
+
+            UnityWebRequest request = UnityWebRequest.Put(uri, payload);
+            request.uploadHandler = new UploadHandlerRaw(Encoding.UTF8.GetBytes(payload));

```

Comment:
This is redundant to `UnityWebRequest.Put` above, as that function [sets the uploadHandlerRaw](https://docs.unity3d.com/Manual/UnityWebRequest-UploadingRawData.html) with the provided data (I would move the `Encoding.UTF8.GetByes` to the Put param though).